### DISCUSSION TITLE
Warn when operation is applied to different types?

### DISCUSSION THREAD

**Discussion by PiN73** [simolus3/drift#1133] [Ideas]
Would it be realistic to implement a warning when a binary operation in .moor file is applied to values having different types?

For example,
```sql
CREATE TABLE user (
    id INT
);

getUser(:id AS TEXT):
    SELECT * FROM user
    WHERE id = :id --<-- [INT] = [TEXT]
    LIMIT 1;
```

or
```sql
CREATE TABLE one (
    id INT,
    val INT
);

CREATE TABLE two (
    id INT,
    val TEXT
);

query:
    SELECT one.val < two.val --<-- [INT] < [TEXT]
    FROM one
    JOIN two
    ON one.id = two.id;
```

**Comment by simolus3** [comment]
With the "obvious" solution (report an error for every mismatching type in a binary expression), I was running into issues when comparing ints and floats (which is perfectly valid). I'll need to see if I can come up with something nicer here. Adding a lot of special cases doesn't seem ideal to me. Type checking is implemented on a graph instead of an AST, so translating that back to helpful error messages isn't simple either.
